Explicitly handle HTTP1 ALPN

moogle19 opened the pull request:
Explicitly handle the `http/1.1` and `http/1.0` ALPN from the client.

mtrudel commented:
Great work! I'm very much in favour of this being merged but there are a few loose ends first:

* We already advertise http/1.1, so #9 should be marked as fixed by this PR (I'll take this one)
* We'll need test cases in https://github.com/mtrudel/bandit/blob/main/test/bandit/initial_handler_test.exs to cover newly addressed permutations (we already cover happy paths for both http/1 and h2 in there, but we'll want to add a 'fails to connect via h2 after negotiating http/1.1' case (the opposite of the case on line 34, essentially).

mtrudel commented:
Fixes #9

moogle19 commented:
@mtrudel 
I added a test case for h2 over HTTP/1.1 and moved the `tls_client` function from the `SimpleH2Client` to `ClientHelpers`  (with `protocols` as a parameter) to reuse it for the test.

mtrudel commented:
Amazing work! Thanks so much!
